[Term Entry] SwiftUI View Modifier: .border() #1511 (#1706)
Codecademy/docs: content/swiftui/concepts/viewmodifier/terms/border/border.md

---
Title: '.border()'
Description: 'Applies a border to a view, which can be customized with a specified style and width.'
Subjects:
  - 'Mobile Development'
  - 'iOS'
Tags:
  - 'SwiftUI'
  - 'SwiftUI ViewModifiers'
  - 'Xcode'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

The **`.border()`** view modifier in SwiftUI can be used to add a border to a view. The border can be customized with a specified style and width.

## Syntax

```pseudo

> View

Reviewer: One more addition would be helpful here about what views this modifier can be used on

Author: Thank you for reviewing my changes @yangc95 
I have updated the documentation to reflect that the .border() view modifier can be applied to any type of View and provided a few examples. Please let me know if there are any further changes or additions you would like me to make.

---
Title: '.border()'
Description: 'Applies a border to a view, which can be customized with a specified style and width.'
Subjects:
  - 'Mobile Development'
  - 'iOS'
Tags:
  - 'SwiftUI'
  - 'SwiftUI ViewModifiers'
  - 'Xcode'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

The **`.border()`** view modifier in SwiftUI can be used to add a border to a view. The border can be customized with a specified style and width.

## Syntax

```pseudo
View

>     .border(_ content: ShapeStyle, width: CGFloat = 1)

Reviewer: For the parameters, can we change `_ content: ShapeStyle` to just `content`

Reviewer: In addition, can we take away the `= 1` for the second parameter as this is included in the explanation below

Author: Thank you for the suggestion, I have added those changes.

```

> This method has one required parameter:
> 
> - `content`: A value that sets the color and style of the border which needs to conform to the `ShapeStyle` protocol, like a `Color` or `HierarchicalShapeStyle`.
> 
> And one optional parameter:
> 
> - `width`: This value determines the width of the border with a default value of 1.

Reviewer: Can we combine these to something like:

```suggestion
This method has one required parameter and one optional parameter:

- `content`: A required value that sets the color and style of the border which needs to conform to the `ShapeStyle` protocol, like a `Color` or `HierarchicalShapeStyle`.
- `width`: This optional value determines the width of the border and defaults to 1.
```

## Example

```swift
var body: some View {
    Text("This view has a blue border.")
        .padding()
        .border(.blue, width: 4)
}
```

> In the above example, we have a Text view that displays the text `"This view has a blue border."` with some padding. The `.border()` modifier is then applied to the Text view, which adds a border around the view.

Reviewer: ```suggestion
In the above example, a `Text` view displays the text `"This view has a blue border."` with some padding. The `.border()` modifier is then applied to the view, which adds a border around it.
```

> In our example, a blue border is specified with a width of 4 points. However, these values can be customized as needed to achieve the desired look for an app.

Reviewer: ```suggestion
In the example, a blue border is specified with a width of 4 points. However, these values can be customized as needed to achieve the desired look for an app.
```

> In our example, a blue border is specified with a width of 4 points. However, you can customize these values as needed to achieve the desired look for your app.

Reviewer 2: ```suggestion
In our example, a blue border is specified with a width of 4 points. However, these values can be customized as needed to achieve the desired look for an app.
```

This will display:

![SwiftUI ViewModifier Border](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-viewmodifier-border.png)

> > **Note**: Keep in mind that the .border() modifier had a `cornerRadius` parameter which has since been deprecated so if you want to round the corners of your border you can use the `overlay()` modifier and have a `RoundedRectangle()` with a `cornerRadius` value instead.

Reviewer: ```suggestion
> **Note**: Keep in mind that the `.border()` modifier had a `cornerRadius` parameter which has since been deprecated. If rounding the corners of the border are needed,  the [`.overlay()`](https://www.codecademy.com/resources/docs/swiftui/viewmodifier/overlay) modifier and a [`RoundedRectangle()`](https://www.codecademy.com/resources/docs/swiftui/views/roundedrectangle) with a `cornerRadius` value can be used instead.
```

> > **Note**: Keep in mind that the .border() modifier had a `cornerRadius` parameter which has since been deprecated so if you want to round the corners of your border you can use the `overlay()` modifier and having a `RoundedRectangle()` with a `cornerRadius` value instead.

Reviewer 2: ```suggestion
> **Note**: Keep in mind that the .border() modifier had a `cornerRadius` parameter which has since been deprecated so if you want to round the corners of your border you can use the `overlay()` modifier and have a `RoundedRectangle()` with a `cornerRadius` value instead.
```
